src/core/search/query_driver.h

```diff
@@ -34,10 +34,17 @@ class QueryDriver {
 
   Parser::location_type location;
 
-  void Add(AstExpr) {
+  void Set(AstNodePtr expr) {
+    expr_ = move(expr);
+  }
+
+  AstNodePtr Get() {
```

Comment:
Ditto about destructive getters :)

Reply:
I suspect it's because Vlad secretly likes rust more than c++.